arcgis-ios-sdk-samples/Maps/Generate offline map/GenerateOfflineMapViewController.swift

```diff
@@ -0,0 +1,209 @@
+        self.offlineMapTask = AGSOfflineMapTask(portalItem: self.portalItem)
+        
+        //setup extent view
+        self.extentView.layer.borderColor = UIColor.red.cgColor
+        self.extentView.layer.borderWidth = 3
+    }
+    
+    private func defaultParameters() {
+        
+        //show progress hud
+        SVProgressHUD.show(withStatus: "Getting default parameters", maskType: .gradient)
+        
+        //default parameters for offline map task
+        self.offlineMapTask.defaultGenerateOfflineMapParameters(withAreaOfInterest: self.frameToExtent()) { [weak self] (parameters: AGSGenerateOfflineMapParameters?, error: Error?) in
+            
+            guard error == nil else {
+                
+                SVProgressHUD.showError(withStatus: error!.localizedDescription, maskType: .gradient)
+                return
+            }
+            
+            //dismiss progress hud
+            SVProgressHUD.dismiss()
+            
+            guard let parameters = parameters else {
+                return
+            }
+            
+            //will need the parameters for creating the job later
+            self?.parameters = parameters
+            
+            //take map offline
+            self?.takeMapOffline()
+        }
+    }
+    
+    private func takeMapOffline() {
+        
+        //create a unique name for the geodatabase based on current timestamp
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        
+        let path = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+        let fullPath = "\(path)/\(dateFormatter.string(from: Date())).geodatabase"
+        
+        self.generateOfflineMapJob = self.offlineMapTask.generateOfflineMapJob(with: self.parameters, downloadDirectory: URL(string: fullPath)!)
+        
+        //add observer for progress
+        self.generateOfflineMapJob.progress.addObserver(self, forKeyPath: "fractionCompleted", options: .new, context: nil)
+        
+        //unhide the progress parent view
+        self.progressParentView.isHidden = false
+        
+        //start the job
+        self.generateOfflineMapJob.start(statusHandler: nil) { [weak self] (result:AGSGenerateOfflineMapResult?, error:Error?) in
+            
+            guard let strongSelf = self else {
+                return
+            }
+            
+            guard error == nil else {
+                SVProgressHUD.showError(withStatus: error!.localizedDescription, maskType: .gradient)
+                return
+            }
+            
+            //disable cancel button
+            strongSelf.cancelButton.isEnabled = false
+            
+            strongSelf.mapView.map = result?.offlineMap
+            
+            //remove KVO observer
+            strongSelf.generateOfflineMapJob.progress.removeObserver(strongSelf, forKeyPath: "fractionCompleted", context: nil)
```

Comment:
looks like we remove observer only when the job finishes. 

What if the user exits the sample before completion?